lib/constructs/callback-function.js

```diff
@@ -0,0 +1,102 @@
+"use strict";
+
+const Types = require("../types");
+const utils = require("../utils");
+
+class CallbackFunction {
+  constructor(ctx, idl) {
+    this.ctx = ctx;
+    this.idl = idl;
+    this.name = idl.name;
+    this.str = null;
+
+    this.requires = new utils.RequiresMap(ctx);
+  }
+
+  generate() {
+    // This is a simplification of https://heycam.github.io/webidl/#web-idl-arguments-list-converting that currently
+    // fits our needs.
+    const argsToES = `
+      for (let i = 0; i < args.length; i++) {
+        args[i] = utils.tryWrapperForImpl(args[i]);
+      }
+      if (args.length < ${this.idl.arguments.length}) {
+        for (let i = args.length; i < ${this.idl.arguments.length}; i++) {
+          args[i] = undefined;
+        }
+      } else if (args.length > ${this.idl.arguments.length}) {
+        args.length = ${this.idl.arguments.length};
+      }
+    `;
+
+    let returnIDL;
+
+    if (this.idl.idlType.idlType === "void") {
+      returnIDL = "";
+    } else {
+      const conv = Types.generateTypeConversion(this.ctx, "returned", this.idl.idlType, [], this.name, "context");
+      this.requires.merge(conv.requires);
+      returnIDL = `
+        ${conv.body}
+        return returned;
+      `;
+    }
+
+    const treatNonObjectAsNull = utils.getExtAttr(this.idl.extAttrs, "TreatNonObjectAsNull");
+    const assertCallable = treatNonObjectAsNull ? "" : `
+      if (typeof value !== "function" && !treatNonObjectAsNull) {
```

Comment:
```suggestion
      if (typeof value !== "function") {
```
The&nbsp;`treatNonObjectAsNull`&nbsp;variable is&nbsp;not&nbsp;defined in&nbsp;the&nbsp;generated&nbsp;file, so&nbsp;it&nbsp;would&nbsp;cause a&nbsp;`ReferenceError` to&nbsp;be&nbsp;thrown.